### PR TITLE
Ports: Add ZSH

### PR DESCRIPTION
**Pull request by Thihup**
Hi. This is my first port to the system, so if I've made something wrong, please tell me :)

To test it out, I tried to run the Oh My Zsh extension (which I use daily), and it was a little harder because the git port doesn't support remote clones yet.

image

### CODE REVIEW

File: Ports/zsh/package.sh [hunk 1]
```diff
@@ -0,0 +1,11 @@
+#!/usr/bin/env -S bash ../.port_include.sh
+port=zsh
+version=zsh-5.8
+files="https://github.com/zsh-users/zsh/archive/refs/tags/${version}.zip zsh-${version}.zip 9fb194253336ae5871913a583e2bdb4db117bed80700539b2524cff606c8ff6f"
```
Comment (gunnarbeutner) [inline Ports/zsh/package.sh:4]: This seems to use an unofficial mirror for the port. Is this intentional?

Comment (gunnarbeutner) [inline Ports/zsh/package.sh:4]:
```suggestion
version=5.8
workdir=zsh-zsh-${version}
files="https://github.com/zsh-users/zsh/archive/refs/tags/zsh-${version}.zip zsh-${version}.zip 9fb194253336ae5871913a583e2bdb4db117bed80700539b2524cff606c8ff6f"
```

**Review by gunnarbeutner — CHANGES_REQUESTED** [review]

File: Ports/zsh/package.sh (hunk 1 above)
Comment (Thihup) [reply Ports/zsh/package.sh:4]: I don't know how to download the source consistently via SourceForge. Any help would be appreciated ;)

File: Ports/zsh/package.sh [hunk 2]
```diff
@@ -0,0 +1,12 @@
+#!/usr/bin/env -S bash ../.port_include.sh
+port=zsh
+version=5.8
+workdir=zsh-zsh-${version}
+files="https://github.com/zsh-users/zsh/archive/refs/tags/zsh-${version}.zip zsh-${version}.zip 9fb194253336ae5871913a583e2bdb4db117bed80700539b2524cff606c8ff6f"
```
Comment (gunnarbeutner) [inline Ports/zsh/package.sh:5]:
```suggestion
version=5.8
files="https://sourceforge.net/projects/zsh/files/zsh/${version}/zsh-${version}.tar.xz zsh-${version}.tar.xz dcc4b54cc5565670a65581760261c163d720991f0d06486da61f8d839b52de27"
```

Can you try if this works for you?

Comment (Thihup) [reply Ports/zsh/package.sh]: It worked, thanks ;)

**Comment by gunnarbeutner** [comment]
According to the checks at https://github.com/SerenityOS/serenity/pull/7808/checks?check_run_id=2766484015 your `package.sh` is missing a new-line at the end of the file. :)

**Review by linusg — CHANGES_REQUESTED** [review]
Please reword the second commit message, instead of this:

> LibC: tgetnum should return -1 if capability is not available

Do something like this:

> LibC: Return -1 from tgetnum() if capability is not available

or:

> LibC: Make tgetnum() return -1 if capability is not available

See https://chris.beams.io/posts/git-commit/#imperative.

Also, ideally the ZSH commit would come last as the other two are prerequisites :)

**Comment by linusg** [comment]
Nice! My favourite shell right after `/bin/Shell` :smile:

**Pull request merged**